pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -22,24 +26,66 @@ type OpenShiftClusterDynamicValidator interface {
 }
 
 // NewOpenShiftClusterDynamicValidator creates a new OpenShiftClusterDynamicValidator
-func NewOpenShiftClusterDynamicValidator(log *logrus.Entry, env env.Interface, oc *api.OpenShiftCluster, subscriptionDoc *api.SubscriptionDocument, fpAuthorizer refreshable.Authorizer) OpenShiftClusterDynamicValidator {
+func NewOpenShiftClusterDynamicValidator(ctx context.Context, log *logrus.Entry, env env.Interface, oc *api.OpenShiftCluster, subscriptionDoc *api.SubscriptionDocument, fpAuthorizer refreshable.Authorizer) (OpenShiftClusterDynamicValidator, error) {
+	var fpClientCred azcore.TokenCredential
+	var spClientCred azcore.TokenCredential
+	var pdpClient remotepdp.RemotePDPClient
+
+	spp := oc.Properties.ServicePrincipalProfile
+
+	if feature.IsRegisteredForFeature(subscriptionDoc.Subscription.Properties, api.FeatureFlagCheckAccessTestToggle) {
+		log.Info("CheckAccess Feature is set")
+		var err error
+		fpClientCred, err = env.FPNewClientCertificateCredential(subscriptionDoc.Subscription.Properties.TenantID)
+		if err != nil {
+			return nil, err
+		}
+
+		spClientCred, err = azidentity.NewClientSecretCredential(subscriptionDoc.Subscription.Properties.TenantID, spp.ClientID, string(spp.ClientSecret), nil)
+		if err != nil {
+			return nil, err
+		}
+
+		aroEnv := env.Environment()
+		pdpClient = remotepdp.NewRemotePDPClient(
+			fmt.Sprintf(aroEnv.AzureRbacPDPEnvironment.Endpoint, env.Location()),
+			aroEnv.AzureRbacPDPEnvironment.OAuthScope,
+			fpClientCred,
+		)
+	}
+
+	tokenClient := aad.NewTokenClient()
+	token, err := tokenClient.GetToken(ctx, log, spp.ClientID, string(spp.ClientSecret), subscriptionDoc.Subscription.Properties.TenantID, env.Environment().ActiveDirectoryEndpoint, env.Environment().ResourceManagerEndpoint)
```

Comment:
do we need to get Token in this function? can this be done lazily later?

Reply:
This is how the existing code works.  I just moved these client creations into the New method rather than them being in the validate method.

Reply:
i'm not arguing the functional or behavioral correctness, but i don't think a New function should make such RPC call.

Reply:
done.  I moved all of them back to `dynamic()`.